controllers/infinispan_controller.go

```diff
@@ -290,6 +290,32 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 		}
 	}
 
+	if infinispan.HasSites() {
+		tunnel := &appsv1.Deployment{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: infinispan.Namespace, Name: infinispan.GetGossipRouterDeploymentName()}, tunnel)
+		if err != nil && errors.IsNotFound(err) {
+			reqLogger.Info("Configuring the Cross-Site Deployment")
+			tunnel, err = r.createGossipRouterTunnel(infinispan)
+			if err != nil {
+				reqLogger.Error(err, "failed to configure new Deployment")
+				return reconcile.Result{}, err
+			}
+			reqLogger.Info("Creating a new Deployment", "Deployment.Name", tunnel.Name)
+			err = r.Client.Create(context.TODO(), tunnel)
```

Comment:
@pruivo another potential issue, if something happens in the cluster, i.e. no resources or anything else, customers never know that Deployment not created correctly. For cluster Pods controller eventually request data from the Pod status and log it into the Infinispan CR status. Seems to need to implement similar logic for the GossipRouter Deployment